src/js/components/Table/Table.js

```diff
@@ -0,0 +1,175 @@
+ *      { name: 'col2', label: 'Column Two', isSortable: true },
+ *      { name: 'col3', label: 'Column Three', render: (column, row) => <a href={`/path/to/${row.id}`}>{ row.col3 }</a> }
+ *    ]
+ *    rows=[
+ *      { id: 1, col1: 'Row 1 Column One', col2: 'Row 1 Column 2', col3: 'Row 1 Column 3' },
+ *      { id: 2, col1: 'Row 2 Column One', col2: 'Row 2 Column 2', col3: 'Row 2 Column 3' },
+ *      { id: 3, col1: 'Row 3 Column One', col2: 'Row 3 Column 2', col3: 'Row 3 Column 3' }
+ *    ] />
+ *
+ */
+
+import React, { PureComponent } from 'react'
+import PropTypes from 'prop-types'
+import { setUniqueId } from '../../utils/helpers'
+import { sortBy } from '../../utils/sort'
+import Icon from '../Icon'
+import Panel from '../Panel'
+import PanelBody from '../PanelBody'
+import PanelRow from '../PanelRow'
+import PanelCell from '../PanelCell'
+import TextLink from '../TextLink'
+
+class Table extends PureComponent {
+  state = {
+    items: [],
+    sortByColumn: null,
+    sortByReverse: false
+  }
+
+  componentWillMount() {
+    const { rows, defaultSort } = this.props
+    const items = setUniqueId(rows)
+    const tableState = defaultSort ? this.sortItems(defaultSort, items) : { items }
+
+    this.setState({ ...tableState })
+  }
+
+  sortItems = (newSort, newItems) => {
+    const { columns } = this.props
+
+    // Check if newSort has a `-`, representing reverse sorting
+    let sortByReverse = newSort.charAt(0) === '-' ? true : false
+    const sortByColumn = sortByReverse ? newSort.substr(1) : newSort
+
+    // If newSort has same name as previous sort - reverse sort direction
+    if (sortByColumn === this.state.sortByColumn) {
+      sortByReverse = !this.state.sortByReverse
+    }
+
+    const { name, sortType, sortFn } = columns.find(c => c.name === sortByColumn)
+
+    const items = sortBy(newItems, { name, sortType, sortFn, isReverse: sortByReverse })
+
+    return { items, sortByColumn, sortByReverse }
+  }
+
+  handleSortClick = e => {
+    e.preventDefault()
+    const { items } = this.state
+    const sortName = e.currentTarget.getAttribute('href')
+    const tableState = this.sortItems(sortName, items)
+
+    this.setState({ ...tableState })
+  }
+
+  renderSortLabel = label => <span className="u-colorInfo u-textNoWrap">{ label }</span>
+
+  renderSortLink = (name, label) => {
+    const { sortByColumn, sortByReverse } = this.state
+    const ascLinkClassName = (name === sortByColumn && !sortByReverse) ? 'u-colorHighlight' : ''
+    const descLinkClassName = (name === sortByColumn && sortByReverse) ? 'u-colorHighlight' : ''
+
+    return (
+      <TextLink location={ name } onClick={ this.handleSortClick } mods="u-flex">
+        { this.renderSortLabel(label) }
+        <div className="u-colorGrey u-fontSizeXs u-spaceLeftXs">
+          <Icon name="up" mods={ `u-block ${ascLinkClassName}` } />
+          <Icon name="down" mods={ `u-block ${descLinkClassName}` } />
+        </div>
+      </TextLink>
+    )
+  }
```

Comment:
This is cool! 

Do we want to build in a right-align option? This is currently used for dollar value columns in LFE (also mocked in the complex table demo in UI Patterns) It basically just adds a couple more classes to the link wrapper: `u-flexJustifyEnd u-spaceNegativeRightSm`. And then the column content cells just get `u-textRight` on each of them. Or do you think these should just get added as additional `mods` when the component is used?

Reply:
That's a good question.  We could set an `alignment` attribute on the columns that could maybe render: right, left, or center.  This would set the appropriate utility methods like `u-textRight or the parent and `u-flexJustifyEnd u-spaceNegativeRightSm` on the link?  W could also automatically try and add the left/right/center justification by checking if the 'u-textRight/Left' mod was passed?

Reply:
Hmm, I'd be fine either way, whatever you think is easier for JS devs. An `alignment` attribute _might_ be easier?